components/camel-soap/src/main/java/org/apache/camel/dataformat/soap/name/ServiceInterfaceStrategy.java

```diff
@@ -170,7 +170,7 @@ private void analyzeServiceInterface(Class<?> serviceInterface) {
                 }
                 inTypeNameToQName.put(ti.getTypeName(), ti.getElName());
             }
-            if (info.getSoapAction() != null && !"".equals(info.getSoapAction())) {
+            if (info.getSoapAction() != null && !info.getSoapAction().isEmpty()) {
```

Comment:
Maybe you could also move `info.getSoapAction()` into a local variable to prevent multiple calls